src/cbackend.cpp

```diff
@@ -4857,8 +4857,15 @@ llvm::Value *SmearCleanupPass::getShuffleSmearValue(llvm::Instruction *inst) con
 
     // Check that the shuffle is a broadcast of the element of the first vector,
     // i.e. mask vector is vector with equal elements of expected size.
-    if (!(mask && (mask->isNullValue() || (shuffleInst->getMask()->getSplatValue() != 0)) &&
-          llvm::isa<llvm::VectorType>(mask->getType()) &&
+
+    if (!(mask &&
+          (mask->isNullValue() ||
+#if ISPC_LLVM_VERSION == ISPC_LLVM_11_0
```

Comment:
``ISPC_LLVM_VERSION == ISPC_LLVM_11_0`` needs to be ``ISPC_LLVM_VERSION >= ISPC_LLVM_11_0``, as it will suddenly break when the trunk switch to 12.0.
Same for other instances this check.